### ISSUE TITLE
Android - Unable to find element using Accessibility ID

### ISSUE THREAD

**Issue by poorvikamahesh** [appium/appium#8337]
Hello,

I am using Appium 1.6.4 and I am trying to find a webView element using the content-desc / Accessibility ID (German language)

I have tried the below -
Accessibility ID - Sonderzeichen (z. B. #$%& amp;) (NOTE: No space was given between "&" and "amp")
Accessibility ID - Sonderzeichen (z. B. #$%&)
xpath - //android.view.View[contains(@content-desc,'Sonderzeichen (z. B. #$%&)')]
xpath - //android.view.View[contains(@content-desc,'Sonderzeichen (z. B. #$%& amp;)')] (NOTE: No space was given between "&" and "amp")

Every time I get a no such element exception -
```

 An element could not be located on the page using the given search parameters. (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 410 milliseconds
For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
Build info: version: '3.0.1', revision: '1969d75', time: '2016-10-18 09:49:13 -0700'
System info: host: 'MACC02R84YFG8WN', ip: '192.168.1.111', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.11.6', java.version: '1.8.0_121'
Driver info: io.appium.java_client.android.AndroidDriver
Capabilities [{app=/Users/DreamMapper/Desktop/poorvika/DreamMapper_dev_April_5.apk, appPackage=com.philips.sleepmapper.root, deviceScreenSize=1080x1920, networkConnectionEnabled=true, warnings={}, databaseEnabled=false, deviceName=087df39b010ad864, fullReset=true, platform=LINUX, deviceUDID=087df39b010ad864, appActivity=com.philips.sleepmapper.activity.SplashScreenActivity, desired={app=/Users/DreamMapper/Desktop/poorvika/DreamMapper_dev_April_5.apk, appPackage=com.philips.sleepmapper.root, appActivity=com.philips.sleepmapper.activity.SplashScreenActivity, platformVersion=6.0.1, platformName=Android, deviceName=Android, fullReset=true, autoAcceptAlerts=true}, platformVersion=6.0.1, webStorageEnabled=false, locationContextEnabled=false, takesScreenshot=true, javascriptEnabled=true, deviceModel=Nexus 5, platformName=Android, deviceManufacturer=LGE, autoAcceptAlerts=true}]
Session ID: 73430cae-48bd-4169-8165-e43e77effb3e
*** Element info: {Using=accessibility id, value=Sonderzeichen (z. B. #$%&amp;)}


```

When I tried to get the page source for the element, I got the below XML for the element
`
<android.view.View index="1" text="" class="android.view.View" package="com.philips.sleepmapper.root" content-desc="Sonderzeichen (z. B. #$%&amp;)" checkable="false" checked="false" clickable="true" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[165,3018][714,1776]" resource-id="" instance="22"/>
`

**Comment by mykola-mokhnach** [comment]
Search by accessibility id is not supported in webviews. See https://github.com/appium/appium/issues/8333 

Regarding xpath -  if there are characters from the special set then they should be escaped properly there. See for example http://stackoverflow.com/questions/12403870/how-to-escape-single-quote-in-xslt-substring-function

**Comment by poorvikamahesh** [comment]
@mykola-mokhnach 
Thank you.

But I am using escape "amp;" for "&". Have updated the query above. I am getting the same error

**Comment by mykola-mokhnach** [comment]
Does this one work?

```
//android.view.View[starts-with(@content-desc,'Sonderzeichen')]
```

**Comment by poorvikamahesh** [comment]
@mykola-mokhnach 

Yes this works. But I need to verify the whole text. 
And the language keeps changing.

So was replacing the text I had to verify in the xpath everytime to verify if that particular text is present in the page or not.
If the exact text is displayed, then the element would be returned. If not, no such element exception would be thrown.

The reason I was doing this is because the xpaths of elements were changing from one device to another. So I cannot give a constant xpath, return the text and then verify the text with expected string

**Comment by mykola-mokhnach** [comment]
I think in your case (check only texts presence) it will be just faster and easier to just get the page source and perform search inside this xml using "normal" xpath queries to the generated document.
